### DISCUSSION TITLE
Install winget with no access to Microsoft Store

### DISCUSSION THREAD

**Discussion by zhengliuX** [microsoft/winget-cli#2890] [Q&A] [answered]
Hi, my computer has no Microsoft Store, is it possible that I install in on Win10 without using the Store?

**Comment by mdanish-kh** [comment, marked as answer]
Yes, you can manually install WinGet through [GitHub Releases](https://github.com/microsoft/winget-cli/releases). You can use this Url that'll download the latest release of WinGet https://github.com/microsoft/winget-cli/releases/latest/download/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle

**Reply by zhengliuX** [reply]
Does this work for Windows10?

**Reply by mdanish-kh** [reply]
Yes, but i forgot to mention that WinGet has a dependency on `Microsoft.UI.Xaml.2.7` and `Microsoft.VCLibs.14.00`. You'll have to manually install these dependencies if you do not have access to MS Store. You can use this PowerShell script to install WinGet and its dependencies if you're on an x64 machine. 

```pwsh
Invoke-WebRequest -Uri https://www.nuget.org/api/v2/package/Microsoft.UI.Xaml/2.7.3 -OutFile .\microsoft.ui.xaml.2.7.3.zip                                                                                                          
Expand-Archive .\microsoft.ui.xaml.2.7.3.zip                                                                         
Add-AppxPackage .\microsoft.ui.xaml.2.7.3\tools\AppX\x64\Release\Microsoft.UI.Xaml.2.7.appx
Add-AppxPackage -Path "https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx"
Add-AppxPackage -Path "https://github.com/microsoft/winget-cli/releases/latest/download/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle"
Remove-Item .\microsoft.ui.xaml.2.7.3\ -r
Remove-Item .\microsoft.ui.xaml.2.7.3.zip
```

**Reply by zhengliuX** [reply]
Thanks a lot! Those errors did occur, so as first I thought it's not for Windows10. Now I know why!

**Reply by zhengliuX** [reply]
Hi, I just noticed that the system is Windows10 enterprise, and is says 
```
Add-AppxPackage command was not found in the module Apps
```  
Is this the restriction of the system?

**Reply by zhengliuX** [reply]
OK, the resaon is I'm using powershell 7, the solution is to run `Import-Module Appx -UseWindowsPowerShell` first

**Reply by mdanish-kh** [reply]
Or you can run this script on "Windows PowerShell" (not to be confused with PowerShell Core / PowerShell 7)

**Reply by zhengliuX** [reply]
Yeah, that works, too!

**Comment by nathanv** [comment]
@mdanish-kh I used the aforementioned script as per https://github.com/microsoft/winget-cli/discussions/2890#discussioncomment-4809180 but I had to adjust it as versions have been updated.
```
Invoke-WebRequest -Uri https://www.nuget.org/api/v2/package/Microsoft.UI.Xaml/2.8.6 -OutFile .\microsoft.ui.xaml.2.8.6.zip                                                                                                          
Expand-Archive .\microsoft.ui.xaml.2.8.6.zip                                                                         
Add-AppxPackage .\microsoft.ui.xaml.2.8.6\tools\AppX\x64\Release\Microsoft.UI.Xaml.2.8.appx
Add-AppxPackage -Path "https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx"
Add-AppxPackage -Path "https://github.com/microsoft/winget-cli/releases/latest/download/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle"
Remove-Item .\microsoft.ui.xaml.2.8.6\ -r
Remove-Item .\microsoft.ui.xaml.2.8.6.zip
```
Is there maybe a script that is more resilient to change? Maybe there is a way to query and parse a manifest for the required dependencies and their versions? Is this script officially documented on a Microsoft documentation page as well?

**Reply by mdanish-kh** [reply]
Yep, the script has since been added to the official MS Learn docs. Any updates to the script will be reflected there hopefully.

https://learn.microsoft.com/en-us/windows/package-manager/winget/#install-winget-on-windows-sandbox
